### PR TITLE
Fix/rgroup multiple labels

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes a bug with multicores.  I was incrementing the RGroup for the atoms matched by the MCS in the second (and third) cores when it should have stayed the same.

Additionally, when removing hydrogens, only look at molecules which have the same rgroups.

And, change the strings to enums where appropriate.

See the new tests for more info.

### CODE REVIEW

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 1]
```diff
@@ -718,9 +763,17 @@ struct RGroupDecompData {
       if (atm == atoms.end()) continue;  // label not used in the rgroup
 
       Atom *atom = atm->second;
-      mappings[indexLabel] = ++count;
+
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: This is the main fix, I wasn't recycling labels, they were always being incremented in the core.  Whups.

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 2]
```diff
@@ -827,27 +879,31 @@ struct RGroupDecompData {
     for (auto &it : best) {
       for (auto rit = it.rgroups.begin(); rit != it.rgroups.end(); ++rit) {
         if (rit->first >= 0) userLabels.insert(rit->first);
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: turns out I don't directly use userLabels.  I'm finding this bit of code could be cleaned up a bit.  However, we do need to keep track of so called indexLabels which probably should be renamed UnassignedLabels which might make more sense.

File: Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp [hunk 3]
```diff
@@ -827,27 +879,31 @@ struct RGroupDecompData {
...
           }
         }
       }
     }
 
+    // Assign final RGroup labels to the cores and propogate these to
+    //  the scaffold
     finalRlabelMapping.clear();
+    
+    int count = 0;
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp]: count really should be maximum assigned rgroup

File: Code/GraphMol/RGroupDecomposition/Wrap/test_rgroups.py [hunk 4]
```diff
@@ -205,6 +205,58 @@ def test_match_only_at_rgroups(self):
...
+                                               asSmiles=True, asRows=False) 
+        self.assertEqual(groups,
+                         {'Core': ['C1COC([*:1])C1', 'C1COC([*:1])C1'],
+                          'R1': ['Cl[*:1].[H][*:1]', 'Cl[*:1].[H][*:1]']})
+        scaffolds = [Chem.MolFromSmarts(x) for x in ("C1NCCC1","C1OCCC1")]
+        groups,unmatched = RGroupDecompose(scaffolds, mols,
+                                           asSmiles=True, asRows=False)
+        self.assertEqual(groups,
+                         {'Core': ['C1CNC([*:1])C1', 'C1COC([*:1])C1', 'C1COC([*:1])C1'],
+                          'R1': ['Cl[*:1].[H][*:1]', 'Cl[*:1].[H][*:1]', 'Cl[*:1].[H][*:1]']})
```
Comment (bp-kelley) [inline Code/GraphMol/RGroupDecomposition/Wrap/test_rgroups.py]: Yay!  It works!

**Comment by bp-kelley** [comment]
Looks like I need to update the c++ tests.

**Comment by bp-kelley** [comment]
@greglandrum formatted and ready to go.

**Comment by greglandrum** [comment]
will merge after the CI passes

**Pull request merged**